CHANGELOG.md

```diff
@@ -6,6 +6,7 @@ Mapbox welcomes participation and contributions from everyone.
 #### Features
 #### Bug fixes and improvements
 - Optimized calls to modify route arrow layer visibility. [#6308](https://github.com/mapbox/mapbox-navigation-android/pull/6308)
+- Provide better java support for `MapboxNavigationApp`. [#6292](https://github.com/mapbox/mapbox-navigation-android/pull/6292)
```

Comment:
should we automate checks if any of public objects' functions or public static functions don't have java interrupt cc @kmadsen

Reply:
Hmm I don't know I just lost some confidence in the @JvmStatic https://mapbox.atlassian.net/browse/NAVSDK-544

https://issuetracker.google.com/issues/158393309

> java.lang.IncompatibleClassChangeError: The method 'com.mapbox.navigation.core.lifecycle.MapboxNavigationApp com.mapbox.navigation.core.lifecycle.MapboxNavigationApp.registerObserver(com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver)' was expected to be of type virtual but instead was found to be of type static (declaration of 'com.mapbox.androidauto.MapboxCarApp' appears in /data/app/~~wWRVVkoksDjAWyl1zcYreA==/com.mapbox.navigation.qa_test_app-grYQDy8r9n-vE5Z1qLuR0Q==/base.apk!classes5.dex)

It changes the method signature where you will risk SEMVER

Reply:
2.8 is the first stable version for `MapboxNavigationApp` so this could be fine. But for other functions that have had previous stable versions - it will change the method signature.